### ISSUE TITLE
Special formatting for `<|`

### ISSUE THREAD

**Issue by avh4** [avh4/elm-format#196]
`<|` is commonly used to introduce multiline expressions, which leads to odd formatted:

```elm
foo x =
    Html.div []
        <| case x of
            Just data ->
                fn data

            Nothing ->
                Html.text ""
```

Instead, let's special case `<|` to be at the end of the line:

```elm
foo x =
    Html.div [] <|
        case x of
            Just data ->
                fn data

            Nothing ->
                Html.text ""
```

For multiline left side:
```elm
foo x =
    (if isLink then
        Html.a []
     else
        Html.div []
    )
    <|
        case x of
            Just data ->
                fn data

            Nothing ->
                Html.text ""
```

**Comment by avh4** [comment]
cc @rtfeldman

**Comment by rtfeldman** [comment]
Sweet! 😃

**Comment by avh4** [comment]
Not sure how things should be handled when there's a chain of operators, with `<|` interspersed among other operators

**Comment by avh4** [comment]
What should this be if we implement this:

```elm
foo x =
    fn
        ++ " "
        <| a 1
        <| b 2
        ++ " "
        <| c 3
```

**Comment by avh4** [comment]
I guess it should be

```elm
foo x =
    fn ++ " " <|
        a 1 <|
        b 2 ++ " " <|
        c 3
```

Are there operators that have higher precedence than `<|` for which this wouldn't make sense?

What about chains of multiline `<|` ?

**Comment by avh4** [comment]
Multiline should maybe be this:

```elm
foo x =
    fn
        ++ " "
    <|
        a 1
    <|
        b 2
            ++ " "
    <|
        c 3
```

**Comment by rtfeldman** [comment]
I would expect it to do this:

```elm
foo x =
    fn
    ++ bar <|
        a 1 <|
            b 2
            ++ baz <|
                c 3
```

**Comment by rtfeldman** [comment]
like `<|` in a multline context means "go down a line and indent, then proceed as normal"

**Comment by avh4** [comment]
I think your example would come out like this, then:

```elm
foo x =
    fn
        ++ bar
    <|
        a 1 <|
            b 2
                ++ baz
            <|
                c 3
```

**Comment by rtfeldman** [comment]
ah, assuming they put a newline *before* the `<|` - got it!

I'm pretty much fine with this...this seems like one of those cases where "you're doing a crazy thing and it's gonna look crazy" should be a fine default until someone drops by with a use case where they're trying to do something reasonable and it ends up doing something undesirable. 😄

**Comment by avh4** [comment]
Okay, now to see if this can actually be done the way that infix operators are parsed...

**Comment by Fresheyeball** [comment]
I'd like to throw out there that instead of special casing `<|` we just give more flexibility to infix operators, and not manipulate them by default. Instead if a operator is found on a new line, elm-format could line it up with other operators found on new lines, but leave them alone otherwise. Right now I find myself frequently adding parenthesis to stop elm-format from making my code less readable. 

If I wrote the following, I'd prefer elm-format not alter it in any way:

```elm
foo x =
    fn ++ bar 
    <| a 1 
    <| b 2 ++ baz 
    <| c 3
```

**Comment by rtfeldman** [comment]
@fresheyeball would you mind opening a separate issue for that? This one has an intentionally narrow scope!

**Comment by Fresheyeball** [comment]
#201

**Comment by rtfeldman** [comment]
Just want to confirm that this would work as follows when anonymous functions are involved:

```elm
foo =
    something blah
        [ another thing <|
            foo bar <|
            \arg1 arg2 ->
                baz
        ]
```
Is the above how I should expect that to be formatted?

**Comment by avh4** [comment]
I would expect the lambda to be indented:

```elm
foo =
    something blah
        [ another thing <|
            foo bar <|
                \arg1 arg2 ->
                    baz
        ]
```

**Comment by rtfeldman** [comment]
Ah, gotcha!

**Comment by avh4** [comment]
Implemented in b7d14c5...8ec981b.

Looks like this:

```elm
foo x =
    Html.div [] <|
        case x of
            Just data ->
                fn data

            Nothing ->
                Html.text ""

foo =
    something blah
        [ another thing <|
            foo bar <|
                \arg1 arg2 ->
                    baz
        ]

foo x =
    fn
        ++ bar
    <|
        a 1 <|
            b 2
                ++ baz
            <|
                c 3
```
